src/app.js

```diff
@@ -1,97 +1,178 @@
 // CAT API
 
-const cats = require('./catsDb.js');
-const breeds = require('./breedsDb.js');
-
 const express = require('express');         // imports express lib module
 const app = express();                     // express() instantiates app 
 const morgan = require('morgan');
 
-const verifyId = (req, res, next) => {
-    const catInQuestion = req.params.id;
-    const catIndex = cats.findIndex(cat => {     // findIndex() iterator 
-        return cat.id == catInQuestion;         // interchangeable with cat['id']
-    }); 
-    if (catIndex !== -1) {
-        req.catIndex = catIndex;
-        next()
+const CatRepository = require('./catsDb.js'); // class instance / object  
+const breeds = require('./breedsDb.js');
+const { deleteCatByIndex } = require('./catsDb.js');
+
+
+// simple JS function 
+const isInvalidString = (value) => {
+    if (typeof value === "string") {
```

Comment:
style nit: you can cut 4 lines down to 1 with the following:

`return typeof value === 'string'`

This also applies to a couple of other places, but use your discretion on whether this is readable or not.

Reply:
If it's two conditions I can't though, right?
Eg: 
if (value !== 'M' || value !== 'F') { ... }